fix: enhance timezone handling for Chile DST transitions

Vansh5632 opened the pull request:
## What does this PR do?

**IANA Zone Conversions**: All timezone operations now use [dayjs.tz()](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) with IANA identifiers instead of manual offset calculations.

**DST-Aware Processing**: The [processWorkingHours()](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) function now:

Uses [createTimeInTimezone()](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) to create availability slots in the organizer's timezone
Respects DST transitions automatically through IANA zone data
Eliminates manual offset adjustments that caused drift
**Consistent Time Handling:**Working hours are applied in the organizer's timezone using IANA logic
- Slot generation respects DST boundaries
- Date boundaries are calculated with DST awareness
**Future-Proof**: The solution works regardless of when/if Chile changes its DST policy, as it relies on IANA timezone data rather than hardcoded assumptions.

- Fixes #23369  (GitHub issue number)
- Fixes CAL-6325 (Linear issue number - should be visible at the bottom of the GitHub issue description)

#### Image Demo (if applicable):

<img width="1358" height="681" alt="Screenshot from 2025-09-03 17-33-45" src="https://github.com/user-attachments/assets/18341f88-c07d-47de-a058-89a68b5d427a" />
<img width="1358" height="681" alt="Screenshot from 2025-09-03 17-33-27" src="https://github.com/user-attachments/assets/7a58fb55-7ee1-4b50-bd8a-04e540a6158a" />


## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

**Added the testcases for validating this thing**

Vansh5632 committed:
fix: enhance timezone handling for Chile DST transitions

Devanshusharma2005 reviewed (commented):
Please address the coderabbit suggestions.

keithwillcode commented:
This should not merge without @emrysal approval

Vansh5632 committed:
fix: improve DST detection for Southern Hemisphere timezones

- Updated isDateInDst() to compare against DST offset rather than assuming January is standard time
- Fixed convertBetweenTimezones() to handle Date/Dayjs objects as absolute instants
- Enhanced timezone validation with more efficient error handling
- Ensures proper DST handling for Chile (America/Santiago) and other Southern Hemisphere zones

All 27 timezone utility tests now pass, including Chile DST scenarios.

kart1ka committed:
Merge branch 'main' into fix/chile-timezone-dst-handling

Vansh5632 commented:
@Devanshusharma2005 is the test cases which are failing is related to the my changes or the changes which are pushed by you

pallava-joshi reviewed (changes requested):
Can you record a Loom video to walk through the changes you made? I’ll help you with the failing test cases, and please make sure to address the coderabbit suggestions as well
